BBallStatsTrack/src/com/example/bballstatstrack/fragments/TeamFragment.java

```diff
@@ -0,0 +1,308 @@
+                        .show();
+                updateTableView();
+            }
+        } );
+        return button;
+    }
+
+    private TextView createEditText( Context context, float weight, String text )
+    {
+        TextView editText = new TextView( context );
+        LayoutParams layoutParams = new LayoutParams( LayoutParams.WRAP_CONTENT, getPixelHeight(), weight );
+        editText.setLayoutParams( layoutParams );
+        editText.setBackgroundResource( R.drawable.cell_border );
+        editText.setText( text );
+        return editText;
+    }
+
+    private TableRow createTableRow( int rowNumber )
+    {
+        TableRow row = new TableRow( getActivity() );
+        row.setId( rowNumber );
+        row.setLayoutParams( new LayoutParams( LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT ) );
+        row.setWeightSum( 16 );
+        TextView leftEditText = createEditText( getActivity(), 1, String.valueOf( mPlayerList.keyAt( rowNumber ) ) );
+        TextView rightEditText = createEditText( getActivity(), 10, mPlayerList.valueAt( rowNumber ) );
+        ImageButton deleteButton = createDeleteButton( rowNumber );
+        row.addView( leftEditText );
+        row.addView( rightEditText );
+        row.addView( deleteButton );
+        return row;
+    }
+
+    private int getPixelHeight()
+    {
+        Resources resources = getResources();
+        return ( int ) TypedValue.applyDimension( TypedValue.COMPLEX_UNIT_DIP, 30, resources.getDisplayMetrics() );
+    }
+
+    private void hideKeyboard()
+    {
+        Activity activity = getActivity();
+        InputMethodManager inputMethodManager = ( InputMethodManager ) activity
+                .getSystemService( Activity.INPUT_METHOD_SERVICE );
+        View focusedView = activity.getCurrentFocus();
+        if( focusedView == null )
+        {
+            return;
+        }
+        inputMethodManager.hideSoftInputFromWindow( focusedView.getWindowToken(), 0 );
+    }
+
+    private void setPlayerNumberEditTextListeners( AlertDialog dialog )
+    {
+        EditText playerNumberText = ( EditText ) dialog.findViewById( R.id.player_number_editText );
+        playerNumberText.addTextChangedListener( new PlayerNumberWatcher( 0, 99 ) );
+    }
+
+    private void setPositiveButtonAction( final AlertDialog dialog )
+    {
+        dialog.getButton( DialogInterface.BUTTON_POSITIVE ).setOnClickListener( new OnClickListener()
+        {
+            @Override
+            public void onClick( View v )
+            {
+                addNewPlayerFromDialog( dialog );
+            }
+        } );
+    }
+
+    private void showNewPlayerDialog()
+    {
+        AlertDialog dialog = buildDialog();
+        dialog.getWindow().setSoftInputMode( WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE );
+        dialog.show();
+        dialog.setCanceledOnTouchOutside( false );
+        setPositiveButtonAction( dialog );
```

Comment:
You can remove this line of code po and instead move the addNewPlayerFromDialog method in the builder.setPositiveButton( android.R.string.ok, new DialogInterface.OnClickListener() po.